File: java_projects/glue_code/commons-graph/src/main/java/org/apache/commons/graph/shortestpath/ShortestPathAlgorithmSelector.java
package org.apache.commons.graph.shortestpath;

import org.graalvm.polyglot.Value;
import org.graalvm.polyglot.PolyglotException;
import org.apache.commons.graph.ContextInitializer;
import org.apache.commons.graph.ExceptionHandler;
import org.apache.commons.graph.IntegrationUtils;
import org.apache.commons.graph.WeightedPath;
import org.apache.commons.graph.weight.OrderedMonoid;
public interface ShortestPathAlgorithmSelector<V, WE, W> {
    <WO extends OrderedMonoid<W>> WeightedPath<V, WE, W> applyingDijkstra(WO weightOperations);
    <WO extends OrderedMonoid<W>> WeightedPath<V, WE, W> applyingBidirectionalDijkstra(
;    <WO extends OrderedMonoid<W>> HeuristicBuilder<V, WE, W> applyingAStar(WO weightOperations);
}

File: data/skeletons/commons-graph/src/main/org/apache/commons/graph/shortestpath/ShortestPathAlgorithmSelector.py
# Imports Begin
from src.main.org.apache.commons.graph.weight.OrderedMonoid import *
from src.main.org.apache.commons.graph.shortestpath.HeuristicBuilder import *
from src.main.org.apache.commons.graph.WeightedPath import *
import typing
import pathlib
from abc import ABC

# Imports End


class ShortestPathAlgorithmSelector(ABC):

    # Class Fields Begin
    # Class Fields End

    # Class Methods Begin
    def applyingDijkstra(
        self, weightOperations: typing.Any
    ) -> WeightedPath[typing.Any, typing.Any, typing.Any]:
        pass

    def applyingBidirectionalDijkstra(
        self, weightOperations: typing.Any
    ) -> WeightedPath[typing.Any, typing.Any, typing.Any]:
        pass

    def applyingAStar(
        self, weightOperations: typing.Any
    ) -> HeuristicBuilder[typing.Any, typing.Any, typing.Any]:
        pass

    # Class Methods End
